core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java

```diff
@@ -63,7 +64,7 @@ public class TreeLock {
   private final List<TreeLockNode> lockNodes;
 
   // TreeLockNode that has been locked.
-  private final Stack<TreeLockNode> heldLocks = new Stack<>();
+  private final Deque<TreeLockNode> heldLocks = new ArrayDeque<>();
```

Comment:
@YxAc 
Thank you for bringing this to our attention. To improve it, I suggest we only need to keep the last 
Index the locked node and then unlock it from the index to 0.

Reply:
> To improve it, I suggest we only need to keep the last
> Index the locked node and then unlock it from the index to 0.

As discussed offline, let's keep the stack for the time being, better readable.